lms/djangoapps/shoppingcart/models.py

```diff
@@ -1606,3 +1609,355 @@ def analytics_data(self):
             data['name'] = settings.PLATFORM_NAME
             data['category'] = settings.PLATFORM_NAME
         return data
+
+
+TRANSACTION_TYPE_PURCHASE = 'p'
+TRANSACTION_TYPE_REFUND = 'r'
+
+
+class PaymentTransaction(TimeStampedModel):
+    """
+    This model will act as a copy of all transaction information that is stored on any Payment Processor. We
+    have a syncronization management command to extract all of the daily reports and store
+    locally in our database
+    """
+
+    # The primary identifier for any transaction as stored in
+    # the payment processor.
+    remote_transaction_id = models.CharField(max_length=64, db_index=True, unique=True)
+
+    # any 'account identified' (aka Merchant Account) associated with this transaction
+    # note, this is an optional field
+    account_id = models.CharField(max_length=32, null=True)
```

Comment:
@wedaly @cdodge , just brainstorming here. We have one concept of Transactions here, and a new one for Invoices in https://github.com/edx/edx-platform/pull/6719 
For reporting and data aggregation it may make sense to have a common model that they extend, to collect general transaction data.